### ISSUE TITLE
“font-smoothing” choices are undocumented/misguided

### ISSUE THREAD

**Issue by matthewpaulthomas** [ubuntudesign/vanilla-design#31]
Vanilla’s `scss/_base_typography.scss` contains [changes to the browser’s default font smoothing](https://github.com/vanilla-framework/vanilla-framework/blob/develop/scss/_base_typography.scss#L106):
```scss
  * {
    // sass-lint:disable no-vendor-prefixes
    // These vendor prefixes are unique and cannot be added by autoprefixer
    -moz-osx-font-smoothing: grayscale;
    -webkit-font-smoothing: antialiased;
    // sass-lint:enable no-vendor-prefixes
    font-smoothing: subpixel-antialiased;
  }
```

Unfortunately, there seems to be no documentation anywhere of why this choice was made: it isn’t mentioned on [the “Font” page](https://docs.vanillaframework.io/en/settings/font-settings), it isn’t explained in the code itself, and even [git blame is no help](https://github.com/vanilla-framework/vanilla-framework/commit/1b012025a516b8921a6f759a02bb76c3137fdaaa) (“Fix up styling structure for v1-dev”).

From what I can find on the Web, people who recommend changing the browser’s default smoothing seem to fall into three categories:
- Those who subjectively prefer one style or another, apparently because they use an OS that defaults to that style (such as iOS using grey-scale smoothing).
- Those who have made mockups in Photoshop, and who are then upset when browsers render a particular font slightly thinner or thicker than Photoshop does.
- Those who are just copying and pasting code from people in one of those other categories, without understanding why. (Some of these people seem to be under the impression that if they don’t customize font smoothing, browsers won’t smooth their text at all!)

I propose that this rule in Vanilla is misguided and inconsistent, and should be removed.

It’s misguided because:
- Its only noticeable effect, when it has any, is to render text in a way that the reader is not used to.
- Vanilla by default uses the Ubuntu font, which has relatively thick strokes and is therefore resilient to small variations in thickness. So the issue of browsers rendering it slightly thicker/thinner than in Photoshop mockups barely matters. (If someone else uses Vanilla with a different font, for which they really think the particular variety of smoothing is important, they can change the smoothing when they change the font.)

And it’s inconsistent in two other ways:
- The `font-smoothing` CSS property does not exist, has never existed, and is not currently proposed for existence. But if it _did_ exist, this rule would be contradicting itself: `-moz-osx-font-smoothing: grayscale` and `-webkit-font-smoothing: antialiased` (when they do anything at all) switch from subpixel to grey-scale smoothing, but `font-smoothing: subpixel-antialiased` would (if it did anything at all) switch from grey-scale to subpixel smoothing!
- If the Ubuntu font is not available, [Vanilla attempts to use your OS’s default font as a fallback](https://github.com/vanilla-framework/vanilla-framework/blob/46e1a98d76a7bfd1d820539acec348d2f31fddc6/scss/_settings_font.scss): `$font-base-family:`…`-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Cantarell"`… (The default fonts of, respectively, macOS, macOS, Windows, Android, KDE systems, and non-Ubuntu Gnome systems.) But it doesn’t make sense to ask the browser to use the OS’s default font, presumably for familiarity, and then to ask that it be rendered inconsistently from how the OS does it.

Alternatively, if there are good reasons to change the font smoothing that outweigh the inconsistency, those reasons are obscure enough that they should be documented in the code and/or the Web site.

**Comment by anthonydillon** [comment]
Investigate @richmccartney with images of cases where this affects you. Look at moving this styling to the `is-dark` pattern.

**Comment by kwm14** [comment]
The reason for allowing `font smoothing` allows us to control the font styling across operating systems.

### With antialiased on
![screen shot 2018-05-15 at 16 11 29](https://user-images.githubusercontent.com/17748020/40065983-f63ae286-585a-11e8-92f5-cd6f03aacabc.png)

### With antialiased off
![screen shot 2018-05-15 at 16 11 45](https://user-images.githubusercontent.com/17748020/40065995-fe1900b4-585a-11e8-9413-fe7581bc843e.png)

**Comment by matthewpaulthomas** [comment]
> The reason for allowing font smoothing

There’s no such thing as “allowing font smoothing” in CSS, since browsers already do it by default. If that wasn’t a typo, it’s an example of what I wrote above: “Some of these people seem to be under the impression that if they don’t customize font smoothing, browsers won’t smooth their text at all!”

> allows us to control the font styling across operating systems.

This is what I was addressing when I wrote: “it doesn’t make sense to ask the browser to use the OS’s default font, presumably for familiarity, and then to ask that it be rendered inconsistently from how the OS does it”.

> With antialiased on
>
> With antialiased off

It’s not clear which of those you prefer. Regardless, that’s an example of what we discussed in the working group meeting last year: the most legitimate use case for overriding the OS’s font smoothing is where it would noticeably affect font weight on dark backgrounds. If so, it could be applied to a class just for dark backgrounds.
